File: 0001.两数相加/solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dic = new Dictionary<int, int>();
        int[] res = new int[2];
        for(int i = 0; i < nums.Length; i++)
        {
            if(dic.ContainsKey(target - nums[i]))
            {
                res[0] = dic[target - nums[i]];
                res[1] = i;
                return res;
            }
            dic.Add(nums[i], i);
        }
        return null;
    }
}

File: 0001.两数相加/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> m;
        vector<int> result;
        for(int i = 0; i < nums.size(); i++){
            int s = target - nums[i];
            if(m.count(s) != 0)
            {
                result.push_back(m[s]);
                result.push_back(i);
                return result;
            }
            m[nums[i]] = i;
        }
        return result;
    }
};